include/zephyr/drivers/can.h

```diff
@@ -173,7 +173,7 @@ struct can_frame {
 	 * @note @kconfig{CONFIG_CAN_RX_TIMESTAMP} must be selected for this
 	 * field to be available.
 	 */
-	uint16_t timestamp;
+	uint64_t timestamp;
```

Comment:
I am still not 100% sold on this change.

We need to document that the full width of the timestamp field is not used by all drivers; some timestamps will wrap around at 32 bit. You also need to look into the format specifiers used (e.g. in the CAN shell). They likely need updating to match the 64 bit size?

Another option could be to mask the upper 32 bit of the timestamp on NXP CANXL, as this seems to be the only device supporting 64 bit timestamps for now?

Reply:
I have tried to survey the market for other CAN controllers supporting 64 bit timestamps, but came up short. Please leave out this change for now.

Reply:
I am sorry, my initial comment said 32 bit, where I meant the existing 16 bits. Please leave out any changes to the timestamp counter bit size from this PR and - if deemed needed - submit a separate PR for this after the S32 CAN driver PR goes in.